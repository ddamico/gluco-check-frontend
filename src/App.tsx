import React from "react";
import { auth } from "./lib/firebase";
import { useAuthState } from "react-firebase-hooks/auth";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import Login from "./pages/Login";
import Landing from "./pages/Landing";
import EditSettings from "./pages/EditSettings";

export default function App() {
  const [user, loading] = useAuthState(auth);
  let Content: React.ReactElement | null = null;
  if (user) {
    Content = <EditSettings user={user} />;
  } else {
    if (!loading) {
      Content = <Landing />;
    }
  }
  return (
    <Router>
      <div>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          {!user && (
            <li>
              <Link to="/login">Login</Link>
            </li>
          )}
          {user && (
            <button
              onClick={() => {
                auth.signOut();
              }}
            >
              Logout
            </button>
          )}
        </ul>

        <hr />
        <Switch>
          <Route exact path="/">
            {Content}
          </Route>
          <Route path="/login">
            <Login />
          </Route>
        </Switch>
      </div>
    </Router>
  );
}
