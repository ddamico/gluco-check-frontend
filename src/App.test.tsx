import React from "react";
import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import App from "./App";

expect.extend(toHaveNoViolations);

test("renders the component", () => {
  const { container } = render(<App />);
  expect(container.firstChild).toMatchInlineSnapshot(`
    <div
      class="App"
    >
      Hey
    </div>
  `);
});

test("has no axe violations", async () => {
  const { container } = render(<App />);
  const axeResults = await axe(container);
  expect(axeResults).toHaveNoViolations();
});
