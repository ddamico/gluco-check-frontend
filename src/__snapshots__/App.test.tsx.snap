// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the component when no user and no loading 1`] = `
<div
  class="makeStyles-root-17"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-23 MuiToolbar-gutters"
    >
      <section
        class="makeStyles-leftToolbar-20"
      >
        <h1
          class="MuiTypography-root makeStyles-navTitle-24 MuiTypography-h6 MuiTypography-colorInherit"
        >
          title
        </h1>
      </section>
      <section
        class="makeStyles-rightToolbar-21"
      >
        <ul
          class="makeStyles-nav-22"
        >
          <li>
            <a
              aria-disabled="false"
              aria-label="boilerplate.faqs"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-home"
              href="/faq"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
        </ul>
      </section>
    </div>
  </header>
  <div
    class="MuiContainer-root makeStyles-container-18 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiPaper-root makeStyles-surface-19 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div>
        Landing
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders the component when user loaded 1`] = `
<div
  class="makeStyles-root-9"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-15 MuiToolbar-gutters"
    >
      <section
        class="makeStyles-leftToolbar-12"
      >
        <h1
          class="MuiTypography-root makeStyles-navTitle-16 MuiTypography-h6 MuiTypography-colorInherit"
        >
          title
        </h1>
      </section>
      <section
        class="makeStyles-rightToolbar-13"
      >
        <ul
          class="makeStyles-nav-14"
        >
          <li>
            <a
              aria-disabled="false"
              aria-label="boilerplate.faqs"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-home"
              href="/faq"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              aria-label="navigation.home"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-home"
              href="/"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              aria-label="navigation.settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-settings"
              href="/settings"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                    transform="scale(1.2, 1.2)"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
          <li>
            <button
              aria-label="navigation.logout"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="logout"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </li>
        </ul>
      </section>
    </div>
  </header>
  <div
    class="MuiContainer-root makeStyles-container-10 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiPaper-root makeStyles-surface-11 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div>
        Welcome
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders the component when user loading 1`] = `
<div
  class="makeStyles-root-1"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-7 MuiToolbar-gutters"
    >
      <section
        class="makeStyles-leftToolbar-4"
      >
        <h1
          class="MuiTypography-root makeStyles-navTitle-8 MuiTypography-h6 MuiTypography-colorInherit"
        >
          title
        </h1>
      </section>
      <section
        class="makeStyles-rightToolbar-5"
      >
        <ul
          class="makeStyles-nav-6"
        >
          <li>
            <a
              aria-disabled="false"
              aria-label="boilerplate.faqs"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-home"
              href="/faq"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
        </ul>
      </section>
    </div>
  </header>
  <div
    class="MuiContainer-root makeStyles-container-2 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiPaper-root makeStyles-surface-3 MuiPaper-elevation1 MuiPaper-rounded"
    />
  </div>
</div>
`;
